### ISSUE TITLE
Todo: ripgrep failed with exit code 1

### ISSUE THREAD

**Issue by JSchrtke** [folke/todo-comments.nvim#13]
I just installed this, running any of the commands yields
```
Todo: ripgrep failed with exit code 1
```
As well as that, none of the items like TODO etc. are highlighted.

I tried this with the following minimal config as well, which yields the same error
```vim
" ### Plugins ###
" TODO
call plug#begin('~/.vim/plugged')
Plug 'folke/todo-comments.nvim'
call plug#end()

" ### Configure todo-comments.nvim ###
lua << EOF
require('todo-comments').setup{}
EOF
```

I am running Ubuntu 20.04.2 LTS, using 
```
ripgrep 11.0.2
-SIMD -AVX (compiled)
+SIMD +AVX (runtime)
```
and
```
NVIM v0.5.0-dev+1318-g61aefaf29
Build type: Release
LuaJIT 2.1.0-beta3
```

**Comment by Shivansh-007** [comment]
Looks like this is a duplicate of #11

**Comment by folke** [comment]
I'm not home right now, but I'll look into this in a couple of hours. It seems that the setup doesnt load. We load everything on VimEnter, or directly if it already entered. Must be something there thats not working as expected

**Comment by folke** [comment]
You could try executing _setup without parameters, right after setup. If that works, then something is wrong with the lazy loading of Todo comments.

**Comment by JSchrtke** [comment]
Just tried that, no change. I don't really have a working Lua setup, otherwise I would do some debugging myself. But I never could get the language server and such to work properly.

**Comment by JSchrtke** [comment]
So, I think I found the problem.
Running `rg TODO:` in a directory that does not contain that search string will return exit code 1. 
This will cause the the call to uv.spawn in line 66 in search.lua to return that as an error in vim, when it actually isn't one. That should probably just be a message along the lines of `No todo's found` or something like that.

You can trigger the same behavior if you have a directory that just generally contains no match for the rg search string.

**Comment by Shivansh-007** [comment]
> You could try executing `require("todo-comments.config")._setup()` without parameters, right after setup. If that works, then something is wrong with the lazy loading of Todo comments.

```lua
E5113: Error while calling lua chunk: ...e/nvim/site/pack/packer/start/packer.nvim/lua/packer.lua:568: ...cker/start/todo-comments.nvim/lua/todo-comments/util.lua:1
8: attempt to index local 'hex' (a nil value)
```
Yup, this is the exact error I get.

**Comment by folke** [comment]
@JSchrtke I've just pushed a fix that only shows an error when exit code is 2.

You said that TODO's aren't highlighted. Are you sure they are followed by a colon? Meaning the following should be highlighted:

* TODO: this is a todo

**Comment by JSchrtke** [comment]
Man, I was just typing out the message for the pull request :P

And yeah, the issue with the non-highlight is indeed the missing colon, but I saw I can just change that in the config. Thanks for the quick fix! 

I think this issue can be closed now, I'm not sure that #11 is actually the same issue.

**Comment by folke** [comment]
@JSchrtke ok, cool. Glad you got it working.
